### PR TITLE
Fix an assertion call during battle shadow rendering

### PR DESCRIPTION
**Pull request by ihhub**
for wide monsters which are at the end of the battlefield

regression from #3911

### CODE REVIEW

File: src/fheroes2/battle/battle_interface.cpp [hunk 1]
```diff
@@ -1646,8 +1646,13 @@ void Battle::Interface::RedrawCover()
             highlightCells.emplace( attackerCell );
 
             if ( _currentUnit->GetTailIndex() != -1 ) {
-                const int tailDirection = _currentUnit->isReflect() ? RIGHT : LEFT;
+                int tailDirection = _currentUnit->isReflect() ? RIGHT : LEFT;
                 const Cell * tailAttackerCell = Board::GetCell( attackerCell->GetIndex(), tailDirection );
+                if ( tailAttackerCell == nullptr ) {
```
Comment (oleg-derevenetz) [inline src/fheroes2/battle/battle_interface.cpp:1651]: I suppose you can use Board::isValidDirection() here as well, this should look more "transparent". But it's up to you :)

Comment (oleg-derevenetz) [inline src/fheroes2/battle/battle_interface.cpp:1651]: I suppose you can use Board::isValidDirection() here as well, this should look more "transparent". But it's up to you :)

Comment (ihhub) [reply src/fheroes2/battle/battle_interface.cpp:1658]: Fixed.

**Review by oleg-derevenetz — COMMENTED** [review]
Hi @ihhub I simplified the logic a bit, please take a look.

**Pull request merged**